Packs/ApiModules/Scripts/HTTPFeedApiModule/HTTPFeedApiModule.py

```diff
@@ -280,34 +291,37 @@ def custom_fields_creator(self, attributes: dict):
         return created_custom_fields
 
 
-def get_no_update_value(response: requests.Response) -> bool:
+def get_no_update_value(response: requests.Response, url: str) -> bool:
     """
     detect if the feed response has been modified according to the headers etag and last_modified.
     For more information, see this:
     https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified
     https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag
     Args:
         response: (requests.Response) The feed response.
+        url: (str) The feed URL (service).
     Returns:
         boolean with the value for noUpdate argument.
         The value should be False if the response was modified.
     """
-
     context = get_integration_context()
-    old_etag = context.get('etag')
-    old_last_modified = context.get('last_modified')
+    old_last_modified = context.get(url, {}).get('last_modified')
 
     etag = response.headers.get('ETag')
     last_modified = response.headers.get('Last-Modified')
 
-    set_integration_context({'last_modified': last_modified, 'etag': etag})
+    context[url] = {'last_modified': last_modified, 'etag': etag}
+    set_integration_context(context)
 
-    if old_etag and old_etag != etag:
-        demisto.debug('New indicators fetched - the ETag value has been updated,'
-                      ' createIndicators will be executed with noUpdate=False.')
+    if response.status_code == 304:
```

Comment:
This logic should be done before saving integration context. When 304 is returned server may respond  with empty etag.